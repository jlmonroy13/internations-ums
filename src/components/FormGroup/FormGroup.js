import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FormField from '../FormField';

class FormGroup extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      description: '',
      file: [],
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChangeFileInput = this.handleChangeFileInput.bind(this);
  }
  handleSubmit(event) {
    event.preventDefault();
    const { file, ...rest } = this.state;
    this.props.createGroup(rest, file);
  }
  handleChange(event) {
    const { name, value } = event.target;
    this.setState(() => ({
      [name]: value,
    }));
  }
  handleChangeFileInput(event) {
    event.preventDefault();
    const { files } = event.target;
    this.setState(() => ({
      file: files[0],
    }));
  }
  render() {
    return (
      <form className="form" onSubmit={this.handleSubmit}>
        <div className="form__field">
          <label className="form__label-file" htmlFor="photo">
            Load photo
          </label>
          <input
            className="form__input"
            type="file"
            id="photo"
            onChange={this.handleChangeFileInput}
            required={true}
          />
        </div>
        <FormField
          type="text"
          id="name"
          name="name"
          placeholder="Group Name"
          onChange={this.handleChange}
          required={true}
        />
        <div className="form__field">
          <label className="sr-only" htmlFor="description">
            Description
          </label>
          <textarea
            className="form__textarea"
            type="text"
            id="description"
            name="description"
            placeholder="This group is for ..."
            onChange={this.handleChange}
            required={true}
          />
        </div>
        <button className="button button--primary push--top" type="submit">
          Add
        </button>
      </form>
    );
  }
}

FormGroup.propTypes = {
  createGroup: PropTypes.func.isRequired,
};

export default FormGroup;
