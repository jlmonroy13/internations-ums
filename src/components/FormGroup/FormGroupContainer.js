import { connect } from 'react-redux';
import { createGroup } from '../../actions/group';
import FormGroup from './FormGroup';

function mapDispatchToProps(dispatch) {
  return {
    createGroup: (data, file) => dispatch(createGroup(data, file)),
  };
}

export default connect(null, mapDispatchToProps)(FormGroup);
