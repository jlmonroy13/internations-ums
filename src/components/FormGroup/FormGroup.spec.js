import React from 'react';
import renderer from 'react-test-renderer';
import FormGroup from './FormGroup';

describe('<FormGroup />', () => {
  it('should render the form group component', () => {
    const createGroup = jest.fn();
    const wrapper = renderer.create(<FormGroup createGroup={createGroup} />);
    expect(wrapper).toMatchSnapshot();
  });
});
