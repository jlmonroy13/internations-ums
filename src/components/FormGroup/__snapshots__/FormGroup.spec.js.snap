// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> should render the form group component 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="form__field"
  >
    <label
      className="form__label-file"
      htmlFor="photo"
    >
      Load photo
    </label>
    <input
      className="form__input"
      id="photo"
      onChange={[Function]}
      required={true}
      type="file"
    />
  </div>
  <div
    className="form__field"
  >
    <label
      className="sr-only"
      htmlFor="name"
    />
    <input
      className="form__input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Group Name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form__field"
  >
    <label
      className="sr-only"
      htmlFor="description"
    >
      Description
    </label>
    <textarea
      className="form__textarea"
      id="description"
      name="description"
      onChange={[Function]}
      placeholder="This group is for ..."
      required={true}
      type="text"
    />
  </div>
  <button
    className="button button--primary push--top"
    type="submit"
  >
    Add
  </button>
</form>
`;
